packages/scripts/src/commands/app-data/convert/parsers/default/default.parser.ts

```diff
@@ -231,21 +232,10 @@ class RowProcessor {
     });
   }
 
-  /**
-   * replace any self references, i.e "hello @row.id" => "hello some_id", @row.text::eng
-   * TODO - should find better long term option that can update based on dynamic value and translations
-   */
+  /** replace any self references, i.e "hello @row.id" => "hello some_id"   */
   private replaceRowSelfReferences() {
-    Object.keys(this.row).forEach((field) => {
-      if (typeof this.row[field] === "string") {
-        const rowReplacements = [...this.row[field].matchAll(/@row.([0-9a-z_:]+)/gim)];
-        for (const replacement of rowReplacements) {
-          const [expression, replaceField] = replacement;
-          const replaceValue = this.row[replaceField];
-          this.row[field] = this.row[field].replace(expression, replaceValue);
-        }
-      }
-    });
+    const context = { row: this.row };
+    return new TemplatedData(context).parse(this.row);
```

Comment:
Legacy methods that searched generically for `@row...` statements have now been refactored and merged into a more general TemplateData class parse method